CS3219-AY2425S1/cs3219-ay2425s1-project-g36
# Account Settings Page; sync AuthContext across browser tabs

Fixes #92 

Added the capability to:
- Update account information (email + username)
- Allow for sync of login and session information across different browser tabs (1 tab logs in, other tab logs in as well)
- Delete account
- Add another functionality for admins to assign/unassign new admins
- Allow changes from external tabs to be updated (e.g. user made admin is reflected automatically through a backend check that runs every 2000ms)

**Testing done:**
Basic delete account functionality:
1. Delete User A account
2. User A should be logged out. Logging in again with User A details should yield User not found error. User A should be able to sign up again as a new account

Integration testing to ensure the delete account works well with the matching service:
1. Enter User A into match queue. Queue should be [ User A ]
3. In new tab, delete User A account
4. User A should be logged out and deleted from match queue as a result (queue should be [])

**Demo screenshots:**
![image](https://github.com/user-attachments/assets/dba57bcc-8331-46ee-b3cd-0fd2d226dd40)

![image](https://github.com/user-attachments/assets/02258d13-a93a-467c-8533-921cf98af400)

![image](https://github.com/user-attachments/assets/83fcaabd-ce04-49f9-8404-2ab470b956a9)
